packages/dds/tree/src/feature-libraries/chunked-forest/codec/schemaBasedEncode.ts

```diff
@@ -122,7 +138,16 @@ export function treeShaper(
 
 		const objectNodeFields: KeyedFieldEncoder[] = [];
 		for (const [key, field] of schema.objectNodeFields ?? []) {
-			objectNodeFields.push({ key, encoder: fieldHandler.shapeFromField(field) });
+			// TODO: Remove this hardcoded check and do this based on either heuristic or schema.
+			const fieldEncoder =
```

Comment:
I will remove this before merging the PR. This is just for review and testing purposes for now.

Reply:
I think you should have a good testing solution before we merge this PR. If you just want something internal not user facing for testing, you can just have a non-exported symbol you can put on a field schema. That would be just about as minimal, but not involve referencing some specific schema here. It would also be way easier to understand for someone looking at the code.

Reply:
Added a callback function to check whether a field should be incrementally encoded. The test uses that to specify fields it wants to be incrementally encoded.